lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -5932,6 +5971,32 @@ LLVMToSPIRVBase::transBuiltinToInstWithoutDecoration(Op OC, CallInst *CI,
                                    transValue(Image, BB),
                                    transValue(Sampler, BB), BB);
   }
+  case OpImageRead:
+  case OpImageSampleExplicitLod:
+  case OpImageWrite: {
+    // Image Op needs handling of SignExtend or ZeroExtend image operands.
+    auto Args = getArguments(CI);
+    SPIRVType *SPRetTy =
+        CI->getType()->isVoidTy() ? nullptr : transScavengedType(CI);
+    auto *SPI = SPIRVInstTemplateBase::create(OC);
+    std::vector<SPIRVWord> SPArgs;
+    for (size_t I = 0, E = Args.size(); I != E; ++I) {
+      if (Args[I]->getType()->isPointerTy()) {
+        Value *Pointee = Args[I]->stripPointerCasts();
+        (void)Pointee;
+        assert((Pointee == Args[I] || !isa<Function>(Pointee)) &&
+               "Illegal use of a function pointer type");
+      }
+      SPArgs.push_back(SPI->isOperandLiteral(I)
+                           ? cast<ConstantInt>(Args[I])->getZExtValue()
+                           : transValue(Args[I], BB)->getId());
+    }
+    if (Args.size() <= getImageOperandsIndex(OC))
+      if (int SignZeroExt = getImageSignZeroExt(CI->getCalledFunction()))
```

Comment:
We should add a check if 1.4 is supported (see `BM->isAllowedToUseVersion(VersionNumber::SPIRV_1_4)`) and either ignore such image operand coming from SPIR-V friendly LLVM IR or even report and error. Tagging @VyacheslavLevytskyy and @vmaksimo to express an opinion here, which way is better.

Reply:
Probably we should ignore rather than report an error, by analogy with https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/93f2783317007343da7e8b00be0a363f2171d77f/lib/SPIRV/libSPIRV/SPIRVInstruction.cpp#L161